### PR TITLE
optimize route table where database and table shared columns

### PR DESCRIPTION
**Pull request by beiwangnull**
Fixes #ISSUSE_ID.
#3366 
Changes proposed in this pull request:
-
-
-

### CODE REVIEW

**Comment by terrymanu** [comment]
Please fix CI first

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngine.java [hunk 1]
```diff
@@ -191,8 +193,38 @@ private boolean isGettingShardingValuesFromHint(final ShardingStrategy shardingS
...
-            result.addAll(routeTables(shardingRule, tableRule, each, tableShardingValues));
+
+            if (null != tableRule.getDatabaseShardingStrategy() && null != tableRule.getTableShardingStrategy()
+                && 0 != tableRule.getDatabaseShardingStrategy().getShardingColumns().size() && 0 != tableRule.getTableShardingStrategy().getShardingColumns().size()
+                && databaseShardingValues.size() > 0 && tableShardingValues.size() > 0
+                && tableRule.getDatabaseShardingStrategy().getShardingColumns().size() == tableRule.getTableShardingStrategy().getShardingColumns().size()
+                && tableRule.getDatabaseShardingStrategy().getShardingColumns().containsAll(tableRule.getTableShardingStrategy().getShardingColumns())) {
+                Collection<Object> relateValueList = Lists.newArrayList();
+
+                ListRouteValue tableRouteValue = (ListRouteValue) tableShardingValues.get(0);
```
[the remark this answers is not in the document]
Comment (beiwangnull) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngine.java:206]: 1、 I didn't expect that RangeValue.

[the remark this answers is not in the document]
Comment (beiwangnull) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngine.java:206]: 2、just care algorithms

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngine.java [hunk 2]
```diff
@@ -191,8 +193,38 @@ private boolean isGettingShardingValuesFromHint(final ShardingStrategy shardingS
...
+                Collection<Object> tablueRouteValues = tableRouteValue.getValues();
+
+                for (Object eachRouteValues : tablueRouteValues) {
+                    RouteValue routeValue = new ListRouteValue(tableRouteValue.getColumnName(), tableRouteValue.getTableName(), Lists.newArrayList(eachRouteValues));
+
+                    Collection<String> reRouteDbRoute = routeDataSources(shardingRule, tableRule, Lists.newArrayList(routeValue));
```
[the remark this answers is not in the document]
Comment (beiwangnull) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngine.java:212]: you are right

**Comment by beiwangnull** [comment]
RangeValue unable to filter the corresponding database，only the determined value can be adapted.

**Pull request closed without merging**